File: youlexuan_parent/youlexuan_manager_web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }

    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    }

    this.save = function (entity) {
        var methodName = "add";
        if (entity.id) {
            methodName = "update";
        }
        return $http.post("../brand/" + methodName + ".do", entity);
    }

    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    }

    this.dele = function (selectIds) {
        return $http.get("../brand/delete.do?ids=" + selectIds);
    }

    this.search = function (searchEntity, page, size) {
        return $http.post("../brand/search.do?page=" + page +"&size=" + size, searchEntity);
    }

    this.selectBrandList = function () {
        return $http.get("../brand/selectBrandList.do");
    }
})

File: youlexuan_parent/youlexuan_sellergoods_interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    void add(TbBrand brand);

    void update(TbBrand brand);

    void delete(long[] ids);

    TbBrand findOne(long id);

    List<Map> selectBrandList();
}
